lib/miq_expression.rb

```diff
@@ -804,9 +804,10 @@ def self.quote(val, typ)
   def self.quote_human(val, typ)
     case typ&.to_sym
     when :integer, :decimal, :fixnum, :float
+      return val if val.kind_of?(Numeric)
       return val.to_i unless val.to_s.number_with_method? || typ == :float
```

Comment:
I feel like this line should have handled your specific case, so I'm not sure why it didn't.

Reply:
unless or.. let me check it out... need a truth table

Reply:
So, both examples are for columns that are floats columns so `typ` is float.  I think I can just remove the `|| typ == :float` and my first guard clause should handle it.  
I believe all floats end up in on line 817/818:

```ruby
else
  val
end
```

which should be the same as the added guard clause

Reply:
Maybe this line is just for strings, but not strings that are like 5.bytes  e.g. `"5"`.  The `|| typ == :float` hurts my brain, though.